pkg/policy/api/rule_validation.go

```diff
@@ -108,23 +108,27 @@ func (e *EgressRule) sanitize() error {
 	return nil
 }
 
-// sanitize validates Kafka rules
+// Sanitize sanitizes Kafka rules
 // TODO we need to add support to check
 // wildcard and prefix/suffix later on.
-func (kr *PortRuleKafka) sanitize() error {
```

Comment:
Hmm.. I seemed to have lost the comment I made this morning. Since we are now including clientID too in the rules, validating that a non-null clientID actually contains valid chars in Sanitize should be added.

We can use KafkaTopicValidChar . For clientID Kafka reference: https://github.com/apache/kafka/blob/f85c18032be5d583c2ca8f9fa1eb1b5ec55e59fa/core/src/main/scala/kafka/consumer/ConsumerConfig.scala  function: validateClientId

Reply:
I don't see much value in constraining the matching capability. If we do so, we will not allow anyone to use strings outside of the spec. I'm happy to do so if it provides any value though.